airthings_ble/parser.py

```diff
@@ -372,24 +370,37 @@ def __init__(
     async def _get_device_characteristics(
         self, client: BleakClient, device: AirthingsDevice
     ) -> AirthingsDevice:
-        # device.identifier = short_address(client.address)
         device.address = client.address
-        for characteristic in device_info_characteristics:
-            try:
-                data = await client.read_gatt_char(characteristic.uuid)
-            except BleakError as err:
-                self.logger.debug("Get device characteristics exception: %s", err)
-                continue
-            if characteristic.name == "hardware_rev":
-                device.hw_version = data.decode(characteristic.format)
-                continue
-            if characteristic.name == "firmware_rev":
-                device.sw_version = data.decode(characteristic.format)
-                continue
-            if characteristic.name == "device_name":
-                device.name = data.decode(characteristic.format)
-            if characteristic.name == "serial_nr":
-                device.identifier = data.decode(characteristic.format)
+        svcs = client.services
+        for service in svcs:
+            for characteristic in service.characteristics:
+                if characteristic.uuid in device_info_characteristics_str:
+                    try:
+                        data = await client.read_gatt_char(characteristic.uuid)
+                    except BleakError as err:
+                        self.logger.warning(
+                            "Get device characteristics exception: %s", err
+                        )
+                        continue
+
+                    if characteristic.uuid == str(CHAR_UUID_HARDWARE_REV):
+                        device.hw_version = data.decode("utf-8")
+                    elif characteristic.uuid == str(CHAR_UUID_FIRMWARE_REV):
+                        device.sw_version = data.decode("utf-8")
+                    elif characteristic.uuid == str(CHAR_UUID_DEVICE_NAME):
+                        device.name = data.decode(characteristic.format)
+                    elif characteristic.uuid == str(CHAR_UUID_SERIAL_NUMBER_STRING):
+                        identifier = data.decode("utf-8")
+                        if identifier != "Serial Number":
+                            device.identifier = identifier
+                    elif characteristic.uuid == str(CHAR_UUID_MODEL_NUMBER_STRING):
+                        device.model_raw = data.decode("utf-8")
+                        device.model = DEVICE_TYPE.get(device.model_raw)
+        if device.name == "":
+            name = f"Airthings {device.model}"
+            if device.identifier != "":
+                name += f" ({device.identifier})"
+            device.name = name
```

Comment:
The most reliable solution is to always do this. This will work on Linux and Mac. And also fix the naming to match names on airthings.com. For example `Wave2` is now rebranded to `Wave Radon`. And `Gen 1` will change from `AT#123456-2900Radon (Serial Number)` (Linux) / `(Serial Number)` (Mac) to `Airthings Wave gen. 1 (123456)`.

_When I write `(Serial Number)`, that is what's being returned, not masked data_
<img width="334" alt="image" src="https://github.com/vincegio/airthings-ble/assets/270292/039e05e9-26f7-441b-8177-f8670ad6b911">


By doing this, I'm afraid it can break current installations. I can see that if I change the name, the HA integration will generate new sensor names for all devices/sensors.

Any input here would be appreciated!

Reply:
Fixed. See my other comment.